lib/foreman_remote_execution/engine.rb

```diff
@@ -0,0 +1,93 @@
+module ForemanRemoteExecution
+  class Engine < ::Rails::Engine
+    engine_name 'foreman_remote_execution'
+
+    config.autoload_paths += Dir["#{config.root}/app/controllers/concerns"]
+    config.autoload_paths += Dir["#{config.root}/app/helpers/concerns"]
+    config.autoload_paths += Dir["#{config.root}/app/models/concerns"]
+
+    # Add any db migrations
+    initializer 'foreman_remote_execution.load_app_instance_data' do |app|
+      app.config.paths['db/migrate'] += ForemanRemoteExecution::Engine.paths['db/migrate'].existent
+    end
+
+    initializer 'foreman_remote_execution.assets.precompile' do |app|
+      app.config.assets.precompile += %w(
+        'template_input.js',
+      )
+    end
+
+    initializer 'foreman_remote_execution.configure_assets', :group => :assets do
+      SETTINGS[:foreman_remote_execution] =
+        {:assets => {:precompile => [
+          'template_input.js',
+        ]}}
+    end
```

Comment:
Do you need both this declaration for generating plugin assets and the declaration on line 14?